src/components/product-tile/ProductTile.tsx

```diff
@@ -49,15 +50,23 @@ const ProductTileContainer = styled.li`
   }
 `;
 
+const postOrderCreateCookie = async (product: Product) => {
+  try {
+    const productOrder = [{ product: product.id, quantity: 1 }];
+    const order = await postOrder(productOrder);
+    createCartCookie(order.id);
+  } catch (err) {
+    console.log(err);
+  }
+};
+
 export const ProductTile: React.FC<{ product: Product }> = ({ product }) => {
   const { t } = useTranslation();
   const { cart, addItem, changeQuantity } = useContext(CartContext);
 
   const handleClick = () => {
-    // TODO: Add to cart
-    const productOrder = [{ product: product.id, quantity: 1 }];
-    postOrder(productOrder);
-    
+    postOrderCreateCookie(product);
+
     const productInCart = cart.find((item) => item.id === product.id);
     productInCart ? changeQuantity(product.id, 1) : addItem(product, 1);
   };
```

Comment:
all the logic in here and `postOrderCreateCookie ` itself should all be in the context. there's nothing special about the product tile or this particular button so don't make it responsible for special things.